### ISSUE TITLE
Pytorch CNN model's low performance issue at evaluation mode

### ISSUE THREAD

**Issue by luwis93choi** [luwis93choi/CNN_VO#3]
CNN model in Pytorch seems to operate very differently under evaluation mode.

Under evaluation mode, the model's performance does not match with the performance characteristics shown during training phase.

As a temporary fix, the model is set on training mode without gradient updates and optimizer operations.

Reference 
: [[BUG] Weird behavior between evaluation and training mode](https://discuss.pytorch.org/t/bug-weird-behavior-between-evaluation-and-training-mode/13297)
: [Performance highly degraded when eval() is activated in the test phase](https://discuss.pytorch.org/t/performance-highly-degraded-when-eval-is-activated-in-the-test-phase/3323)
: [Pytorch eval 성능 문제](https://gldmg.tistory.com/124)

**Comment by luwis93choi** [comment]
Conclusion : This is due to serious model overfitting / No bug in Pytorch evaluation

Using convolutional neural network model with 4 ~ 5 layers is not enough for visual odometry problem.
Estimating pose change from the stack of 2 consecutive images requires much deeper CNN structure.

CNN with 8 convolutional layers and 3 fully connected layer does not overfit in estimating pose change from the stack of 2 consecutive images.
